test/state/host.cpp

```diff
@@ -0,0 +1,368 @@
+// evmone: Fast Ethereum Virtual Machine implementation
+// Copyright 2022 The evmone Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "host.hpp"
+#include "rlp.hpp"
+
+namespace evmone::state
+{
+bool Host::account_exists(const address& addr) const noexcept
+{
+    const auto* const acc = m_state.get_or_null(addr);
```

Comment:
Wouldn't this also put it into the hot list?

Reply:
This will not (it will not create an account if it doesn't exist). And I believe it should not - this is related to old call cost penalty check. But we should have a test which check if this does not affect account access list.